src/Microsoft.ML.Dnn/DnnCatalog.cs

```diff
@@ -144,7 +144,7 @@ public static class DnnCatalog
         {
             var options = new ImageClassificationEstimator.Options()
             {
-                ModelLocation = arch == Architecture.ResnetV2101 ? @"resnet_v2_101_299.meta" : @"InceptionV3.meta",
+                ModelLocation = ModelLocation[arch],
```

Comment:
>ModelLocation = ModelLocation[arch], [](start = 16, length = 36)

Not your fault but since we are here might as well correct it. ModelLocation is not something that we need as an input from the user hence it should not be part of the options class. PLease remove this and move it as a property of ImageClassificationTransformer but not need to serialize it. This would mean you will need to move download code from lines 172 to 205 into ImageClassification constructor and specifically when we are not loading the model from disk, i,.e we are training the model for the first i.e within if (loadModel == false) case. This also means we can move ImagePreprocessingSize and ModelLocation constants into ImageClassificationTransformer since they are not really a property that needs to be set for initializing ImageClassification Estimator or something we need to expose publically.

Reply:
Since this change is not realted to the current PR adding MobilenetV2 support, I will make this change in the PR fixing [https://github.com/dotnet/machinelearning/issues/4307](https://github.com/dotnet/machinelearning/issues/4307).

---
In reply to: [337218608](https://github.com/dotnet/machinelearning/pull/4351#discussion_r337218608) [](ancestors = 337218608)

Reply:
ok, thanks @ashbhandare !